using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Vulpine.Core.Data.Exceptions
{
    /// <summary>
    /// Somtimes there are branches of code that should never be taken during normal
    /// excicution, yet must be included for sake of completness. These branches throw
    /// this exception to indicate something has gone wrong. Usualy the throwing of this
    /// exception indicates something is wrong with the library itself, and vary rarely
    /// a system malfunction. It is not an exception that should be caught.
    /// <remarks>Last Update: 2015-10-08</remarks>
    /// </summary>
    public class UnreachableExcp : DataStruckExcp
    {
        /// <summary>
        /// The default message generated by an unrechable exception.
        /// </summary>
        public const string MSG = "This exception should never be thrown during "
            + "normal excicution. The rasing of this exception indicates "
            + "somehting is wrong with the library itself.";

        /// <summary>
        /// Constructs a new Unrechable Exception 
        /// </summary>
        public UnreachableExcp() : base() { }

        /// <summary>
        /// Generates a custom message, indicating the nature of the exception.
        /// </summary>
        public override string Message
        {
            get { return MSG; }
        }
    }
}
